base/tests/unittests/com/google/idea/blaze/base/lang/buildfile/lexer/BlazeLexerTest.java

```diff
@@ -63,28 +63,28 @@ public void testBasics4() throws Exception {
     assertEquals("", names(tokens("")));
     assertEquals("", names(tokens("# foo")));
     assertEquals("INT INT INT INT", names(tokens("1 2 3 4")));
-    assertEquals("INT DOT INT", names(tokens("1.234")));
+    assertEquals("FLOAT", names(tokens("1.234")));
     assertEquals(
         "IDENTIFIER LPAREN IDENTIFIER COMMA IDENTIFIER RPAREN", names(tokens("foo(bar, wiz)")));
   }
 
   @Test
   public void testIntegersAndDot() throws Exception {
-    assertEquals("INT(1) DOT INT(2345)", values(tokens("1.2345")));
+    assertEquals("FLOAT(1.2345)", values(tokens("1.2345")));
 
-    assertEquals("INT(1) DOT INT(2) DOT INT(345)", values(tokens("1.2.345")));
+    assertEquals("FLOAT(1.2) FLOAT(0.345)", values(tokens("1.2.345")));
 
-    assertEquals("INT(1) DOT INT(0)", values(tokens("1.23E10")));
-    assertEquals("invalid base-10 integer constant: 23E10", lastError);
+    assertEquals("FLOAT(1.23E10)", values(tokens("1.23E10")));
 
-    assertEquals("INT(1) DOT INT(0) MINUS INT(10)", values(tokens("1.23E-10")));
-    assertEquals("invalid base-10 integer constant: 23E", lastError);
+    assertEquals("FLOAT(0.0)", values(tokens("1.23E")));
+    assertEquals("invalid float constant: 1.23E", lastError);
 
+    assertEquals("FLOAT(1.23E-10)", values(tokens("1.23E-10")));
     assertEquals("DOT INT(123)", values(tokens(". 123")));
-    assertEquals("DOT INT(123)", values(tokens(".123")));
+    assertEquals("FLOAT(0.123)", values(tokens(".123")));
     assertEquals("DOT IDENTIFIER(abc)", values(tokens(".abc")));
 
-    assertEquals("IDENTIFIER(foo) DOT INT(123)", values(tokens("foo.123")));
+    assertEquals("IDENTIFIER(foo) FLOAT(0.123)", values(tokens("foo.123")));
```

Comment:
I'm not sure the spec covers this, but I think an identifier that ends in numbers should still parse the same way:

```suggestion
    assertEquals("IDENTIFIER(foo) FLOAT(0.123)", values(tokens("foo.123")));
    assertEquals("IDENTIFIER(foo456) FLOAT(0.123)", values(tokens("foo567.123")));
```

Reply:
The spec says:

> Each token is formed from the longest sequence of characters that would form a valid token of each kind.

(i.e., the "maximal munch") so it does cover this.

Test added, passes :)